client/controller/index.web.js

```diff
@@ -24,25 +23,64 @@ export { setSection, setUpLocale } from './shared.js';
 
 const user = userFactory();
 
-export const ReduxWrappedLayout = ( { store, primary, secondary, redirectUri } ) => (
-	<ReduxProvider store={ store }>
-		{ getCurrentUser( store.getState() )
-			? <Layout primary={ primary }
-				secondary={ secondary }
-				user={ user }
-				nuxWelcome={ nuxWelcome }
-				translatorInvitation={ translatorInvitation }
-			/>
-			: <LayoutLoggedOut
-				primary={ primary }
-				secondary={ secondary }
-				redirectUri={ redirectUri }
-			/>
+/**
+ * Fetch the correct Layout based on whether or not the user is logged-in.
+ *
+ * @param {bool} loggedIn true if the user is logged-in, false otherwise
+ * @returns {Promise} a Promise that resolves to the Layout component
+ */
+function getLayout( loggedIn ) {
+	return new Promise( ( resolve ) => {
+		if ( loggedIn ) {
+			// If user is logged in, async load the main layout
+			require.ensure( [], () => {
```

Comment:
@nb I think, however, the related [`asyncRequire`](https://github.com/Automattic/wp-calypso/blob/0ca65f4bddbefb8375f33fe7cbd5ddf736604597/server/bundler/babel/babel-plugin-transform-wpcalypso-async/test/index.js#L52) function would work. This would be beneficial since it falls back to a static `require` for a non-async environment. I'm going to try using that instead of `require.ensure`.